src/library/scala/util/Try.scala

```diff
@@ -202,13 +202,22 @@ sealed abstract class Try[+T] extends Product with Serializable {
 }
 
 object Try {
-  /** Constructs a `Try` using the by-name parameter.  This
-   * method will ensure any non-fatal exception is caught and a
-   * `Failure` object is returned.
+  /** Constructs a `Try` using the by-name parameter as a result value.
+   *
+   *  The evaluation of `r` is attempted once.
+   *
+   *  Any non-fatal exception is caught and results in a `Failure`
+   *  that holds the exception.
+   *
+   *  @param r the result value to compute
+   *  @return the result of evaluating the value, as a `Success` or `Failure`
    */
   def apply[T](r: => T): Try[T] =
-    try Success(r) catch {
-      case NonFatal(e) => Failure(e)
+    try {
+      val r1 = r
+      Success(r1)
```

Comment:
`Success` doesn't have a by-name parameter, so this should be the same as before?

Reply:
Ah, I saw the other discussion now https://github.com/scala/scala/pull/10600#issuecomment-1804258816.